packages/web-components/src/components/masthead/masthead-l1.ts

```diff
@@ -60,19 +60,25 @@ function handleDropdownClose(event: FocusEvent | KeyboardEvent) {
       }
       case 'focusout': {
         const { relatedTarget } = event as FocusEvent;
-        if (
-          !relatedTarget ||
-          !currentTarget ||
-          !(
-            (relatedTarget as Node).compareDocumentPosition(
-              currentTarget as Node
-            ) & 8
-          )
-        ) {
+        const hasUnknownTargets = !relatedTarget || !currentTarget;
```

Comment:
this is saying `hasUnknownTargets` is literally everything except `relatedTarget` or `currentTarget`?

Reply:
Or is this a boolean...?

Reply:
In [FocusEvents](https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent) there is a `currentTarget` property which points to the element that _triggered_ the event. This is the element that lost focus in `focusout` events and the one that gained focus on `focusin` events.

They also have a special `relatedTarget` property that points to the "other" element in the equation.

In `focusout` events, focus leaves `currentTarget` and goes to `relatedTarget`.

In `focusin` events, focus leaves `relatedTarget` and goes to `currentTarget`.

The `hasUnknownTargets` variable here is a boolean that checks if either is falsy. The most common case for this is someone has navigated away from the page. In this case, focus has left the dropdown and gone to `undefined`.